File: packages/sdk-java/src/main/java/com/zenvio/sdk/model/SendResponse.java
package com.zenvio.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class SendResponse {
    private boolean success;
    private String messageId;
    private String jobId;
    private String error;

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessageId() {
        return messageId;
    }

    public void setMessageId(String messageId) {
        this.messageId = messageId;
    }

    public String getJobId() {
        return jobId;
    }

    public void setJobId(String jobId) {
        this.jobId = jobId;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public static SendResponse error(String message) {
        SendResponse response = new SendResponse();
        response.setSuccess(false);
        response.setError(message);
        return response;
    }
}

File: packages/sdk-php/src/Models/SendResponse.php
<?php

namespace Zenvio\Models;

class SendResponse
{
    public bool $success;
    public ?string $messageId = null;
    public ?string $jobId = null;
    public ?string $error = null;

    public function __construct(array $data)
    {
        $this->success = $data['success'] ?? false;
        $this->messageId = $data['messageId'] ?? null;
        $this->jobId = $data['jobId'] ?? null;
        $this->error = $data['error'] ?? $data['message'] ?? null;
    }

    public static function fromError(string $message): self
    {
        return new self(['success' => false, 'error' => $message]);
    }
}
